packages/core/src/generators/utils/generate-project.ts

```diff
@@ -55,12 +55,16 @@ export async function normalizeOptions(
   client?: DotNetClient,
   projectType?: ProjectType,
 ): Promise<NormalizedSchema> {
-  const name = names(options.name).fileName;
+  const name =
+    options.pathScheme === 'nx' ? names(options.name).fileName : options.name;
   const className = names(options.name).className;
   const projectDirectory = options.directory
     ? `${names(options.directory).fileName}/${name}`
     : name;
-  const projectName = projectDirectory.replace(/\//g, '-');
+  const projectName =
+    options.pathScheme === 'nx'
+      ? projectDirectory.replace(/\//g, '-')
+      : projectDirectory;
```

Comment:
This should probably be something like:
```suggestion
  const projectName =
    options.pathScheme === 'nx'
      ? projectDirectory.replace(/\//g, '-')
      : `${names(options.directory).className}.${className}`;
```

I don't really like the way the code looks / readability at this point, but that's more of a comment on the overall state of this normalization method... If you can clean up this little section some it would be appreciated.

Reply:
We should decide what we want the project name to be in workspace.json because if I am not mistaked, the projectName is what is used to add the project to NX.  

So I see a couple of cases that could be NX pathScheme and dotnet.
1- nx g @nx-dotnet/core:library SomeCompany.Foobar --directory=somefolder ...
2- nx g @nx-dotnet/core:library SomeCompany.Foobar ...

In other words, with and without directory

NX Scheme  with directory: somefolder-some-company.foobar,  <directory>-<name>
NX Scheme without directory: some-company.foobar, <name>

And I would assume the following for dotnet scheme
Dotnet Scheme with directory: Somefolder.SomeCompany.Foobar
Dotnet Scheme without directory: SomeCompany.Foobar

What is the reason for adding the directory to the front of the  name?  Just to namespace it?

Feel free to message me directory to discuss.  I am not sure if I was clear here.  I want to get clarity on all cases before I rip this methods apart and try make it cleaner.   That includes making the changes for projectName.

Reply:
Yeah, adding the directory is mainly to namespace it, but also to maintain consistency w/ the other Nx generators. The csproj name matches the generated project name, and w/ the nx first party generators when you specify --directory, it is prepended to the project name.